### PR TITLE
chore: Upgrade `revm`

### LINKED ISSUE #5490: Upgrade `revm` in `reth` to a version including the `Canyon` hardfork

### PR DESCRIPTION
**Pull request by clabby**
## Overview

> Re-opened version of #5489, base of the PR stack for the Canyon hardfork in `op-reth`.

Upgrades `revm` to the head commit (as of Nov 20, 2023), which includes the `Canyon` hardfork definition.

**Metadata**
closes https://github.com/paradigmxyz/reth/issues/5490

### CODE REVIEW

File: crates/revm/revm-inspectors/src/tracing/js/mod.rs [hunk 1]
```diff
@@ -101,7 +103,7 @@ impl JsInspector {
             .cloned()
             .ok_or(JsInspectorError::ResultFunctionMissing)?;
         if !result_fn.is_callable() {
-            return Err(JsInspectorError::ResultFunctionMissing)
+            return Err(JsInspectorError::ResultFunctionMissing);
```
Comment (mattsse) [inline crates/revm/revm-inspectors/src/tracing/js/mod.rs:106]: please no more trailing semicolons 😭

File: crates/revm/revm-inspectors/src/stack/maybe_owned.rs [hunk 2]
```diff
@@ -85,86 +90,92 @@ where
...
         }
     }
 
     fn call(
         &mut self,
-        data: &mut EVMData<'_, DB>,
+        context: &mut EvmContext<'_, DB>,
         inputs: &mut CallInputs,
-    ) -> (InstructionResult, Gas, Bytes) {
+    ) -> Option<(InterpreterResult, Range<usize>)> {
```
Comment (mattsse) [inline crates/revm/revm-inspectors/src/stack/maybe_owned.rs:117]: this api is kinda weird

**Review by mattsse — CHANGES_REQUESTED** [review]
need to check, new revm api.

in the meantime please remove trailing ;

File: crates/revm/revm-inspectors/src/tracing/js/mod.rs (hunk 1 above)
Comment (clabby) [reply crates/revm/revm-inspectors/src/tracing/js/mod.rs:106]: I'm sorry - there's a bug in `cargo +nightly fmt --all` it seems on darwin where it's not respecting the `rustfmt.toml`, gotta go in and do it manually 🫠

File: crates/revm/revm-inspectors/src/stack/maybe_owned.rs (hunk 2 above)
Comment (clabby) [reply crates/revm/revm-inspectors/src/stack/maybe_owned.rs:117]: Yeah I'm not convinced the ranges being passed are correct... Need to read into what it signifies

**Pull request closed without merging**